docs/src/tari_script_no_op_vulnerability.md

```diff
@@ -0,0 +1,82 @@
+transaction negotiation they commit to the script hash \\( \sigma_b \\), and the Public UTXO script offset, \\( K_{Ob} \\), which is provided by Alice. These are committed 
+to in the construction of 
+
+\\( \beta_b = H(\sigma_b || F_b || K_{Ob}) \\).
+
+\\( \beta_b \\) is then used to create the modified commitment 
+
+\\( \hat{C}_b = (k_b + \beta_b) \cdot G + v_b \cdot H \\)
+
+
+used to produce the range proof \\( RP_b \\). The assumption is that only Alice has access to the private key, \\( k_{Ob} \\) used to produce the Public UTXO script 
+offset for Bob script. 
+
+Once Bob has replied with their part of the transaction negotiation Alice completes the transaction by producing the complete MW transaction kernel signature and 
+two more operations for TariScript aspects of the transaction. Firstly, Alice constructs a script signature (\\( s_{sa}, R_{sa} \\)) for spending \\( C_a \\) by signing 
+\\( H( R_{sa} || \alpha_a || \theta_a || h) \\) with her private script key \\( k_{sa} \\). Secondly, Alice constructs the transaction's script offset
+
+\\( \gamma =  k_{sa} - k_{Ob}U_b \\).
+
+This offset can ostensibly only be constructed by someone who has knowledge of \\( k_{sa} \\) and \\( k_{Ob} \\). \\( k_{sa} \\) is the private script key whose public value, 
+\\( K_{sa} \\) is the public key that will be left on the stack when Alice's input script, \\( \alpha_a \\), is executed using Alice's input data \\( \theta_a \\).
+
+However, the public script key-pair \\( k_{sa} / K_{sa} \\) is purely a product of the script \\( \alpha_a \\) and the input data \\( \theta_a \\) and in the case of the NO_OP script
+there are no operations in the script that use data that Bob is not aware of. This means that Bob could choose a new set of input data \\( \hat{\theta_a} \\) which contains a public script
+key of his choosing, \\( \hat{K_{sa}} \\) to which he knows the private key \\( \hat{k_{sa}} \\). The execution of the NO_OP script for Alice's input will now leave \\( \hat{K_{sa}} \\) on 
+the stack. Bob can choose a new nonce \\( \hat{R_{sa}} \\) and then produce a valid script signature (\\( \hat{s_{sa}}, \hat{R_{sa}} \\)) for the challenge 
+\\( H( \hat{R_{sa}} || \alpha_a || \hat{\theta_a} || h) \\).
+
+Bob can also now produce a new transaction script offset while choosing a new UTXO script offset for his UTXO, \\( \hat{k_{Ob}} \\). This means he can modify the script attached 
+to his UTXO, \\( \hat{\sigma_{b}} \\), calculate a new modified commitment and range proof \\( \hat{RP_{b}} \\) for his new script and commit to it in a new transaction script offset
+
+\\( \hat{\gamma} =  \hat{k_{sa}} - \hat{k_{Ob}} \hat{U_b} \\).
+
+This attack is only possible if Alice did not have a change output in the transaction because in that case Bob would not know the private script offset for the change output, \\( k_{oc} \\), to produce 
```

Comment:
```suggestion
This attack is only possible if Alice did not have a change output that he does not know the blinding factor for in the transaction because in that case Bob would not know the private script offset for the change output, \\( k_{oc} \\), to produce 
```

This attack is only open are ALL inputs are NO_OP and Bob knows the blingfactor for ALL outputs.

Reply:
lol, I am not sure how to measure the `bling` of an output. 

But I will update the wording to include the idea that all the inputs need to be NO_OP and the change output. I will briefly mention that its all outputs but I prefer to keep the cleaner paradigm of 1 change output and 1 receiver output are the outputs that can be in a Tx.